app/models/assignment.rb

```diff
@@ -10,14 +10,9 @@ class Assignment < ApplicationRecord
 
   validates_presence_of :exercise, :submitter
 
-  [:expectation_results, :test_results, :query_results].each do |field|
-    serialize field
-    define_method(field) { self[field]&.map { |it| it.symbolize_keys } }
-  end
+  delegate :language, :name, to: :exercise
 
-  delegate :language, :name, :visible_success_output?, to: :exercise
-  delegate :output_content_type, to: :language
-  delegate :should_retry?, to: :status
+  alias_attribute :status, :submission_status
```

Comment:
Nice, I din't know about this macro. Since we are using it, can we just keep using the `status` name in queries and `to_json`?